test-app/src/api-test.ts

```diff
@@ -0,0 +1,172 @@
+import {
+  App,
+  aws_apigateway,
+  aws_dynamodb,
+  aws_logs,
+  Stack,
+} from "aws-cdk-lib";
+import {
+  ApiIntegrations,
+  ExpressStepFunction,
+  Function,
+  SyncExecutionSuccessResult,
+  Table,
+} from "functionless";
+
+export const app = new App();
+
+const stack = new Stack(app, "api-test-app-stack");
+
+const restApi = new aws_apigateway.RestApi(stack, "api", {
+  restApiName: "api-test-app-api",
+});
+
+interface FnRequest {
+  pathParameters: {
+    num: number;
+  };
+  queryStringParameters: {
+    str: string;
+  };
+  body: {
+    bool: boolean;
+  };
+}
+
+const fn = new Function(
+  stack,
+  "fn",
+  async (event: { inNum: number; inStr: string; inBool: boolean }) => ({
+    fnNum: event.inNum,
+    fnStr: event.inStr,
+    fnBool: event.inBool,
+    nested: {
+      again: {
+        num: 123,
+      },
+    },
+  })
+);
+
+const fnResource = restApi.root.addResource("fn").addResource("{num}");
+const fnIntegration = ApiIntegrations.aws({
```

Comment:
I think this needs to be a class or else it's inconsistent with the rest of the framework. Did you say there are type errors with the class? I'd like to take a look at that.

Reply:
it's set up so you can use a class, but you don't get type inference. checkout out this branch and then compare to this:

```
const fnIntegration2 = new AwsApiIntegrations({
  request: (req: FnRequest) =>
    fn({
      inNum: req.pathParameters.num,
      inStr: req.queryStringParameters.str,
      inBool: req.body.bool,
    }),
  response: (resp) => ({         // <-----------------`resp` here will be typed as `any`
    resultNum: resp.fnNum,
    resultStr: resp.fnStr,
    nested: resp.nested.again.num,
  }),
  errors: {
    400: () => ({ msg: "400" }),
  },
});
```

It works in 4.7 with the method because of https://github.com/microsoft/TypeScript/pull/48538